src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogSummaryMailGenerator.java

```diff
@@ -102,16 +105,32 @@ public String prepareMailContent(int numberOfDaysBackInThePast, long issueNumber
         ).collect(Collectors.toList());
     }
 
-    private Map<Blog, String> getBlogAndItsHomepage(List<Blog> blogsAddedSinceLastNewsletter) {
-        return blogsAddedSinceLastNewsletter.stream().collect(Collectors.toMap(
-            Function.identity(),
-            blog -> getBlogHomepageFromRss(blog.getRss()))
-        );
+    private Map<Blog, HomepageUrl> getBlogAndItsHomepage(
+        List<Blog> blogsAddedSinceLastNewsletter) {
+        return blogsAddedSinceLastNewsletter.stream()
+            .map(blog -> Pair.of(blog,
+                new HomepageUrl(getBlogHomepageFromRss(blog.getRss()))))
+            .collect(Collectors.toMap(Pair::getLeft, Pair::getRight)
+            );
     }
 
-    private String getBlogHomepageFromRss(String rss) {
-        SyndFeed syndFeed = syndFeedFactory.createFor(rss);
-        return syndFeed.getLink();
+    private Optional<String> getBlogHomepageFromRss(String rss) {
+        return syndFeedFactory.createFor(rss).map(SyndFeed::getLink);
+    }
+
+    public static class HomepageUrl {
```

Comment:
Just a convention (because homepage and home page are both correct words now in English I think), I just was using the old but sure thing, we can switch this to treating home page as two words.